src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,1638 @@
+            losses=losses,
+        )
+
+    def forward(self, pixel_values: Tensor, targets: Optional[Dict[str, Tensor]] = None) -> MaskFormerOutput:
+        image_features, pixel_embeddings = self.pixel_level_module(pixel_values)
+        queries = self.transformer_module(image_features)
+        outputs: Dict[str, Tensor] = self.segmentation_module(queries, pixel_embeddings)
+
+        loss_dict: Dict[str, Tensor] = {}
+        loss: Tensor = None
+        if targets:
+            loss_dict = self.get_loss_dict(outputs, targets)
+            loss = self.get_loss(loss_dict)
+        else:
+            # upsample the masks to match the inputs' spatial dimension
+            outputs["pred_masks"] = upsample_like(outputs["pred_masks"], pixel_values)
+
+        return MaskFormerOutput(**outputs, loss_dict=loss_dict, loss=loss)
+
+    def get_loss_dict(self, outputs: Dict[str, Tensor], targets: Dict[str, Tensor]) -> Dict[str, Tensor]:
+        loss_dict: Dict[str, Tensor] = self.criterion(outputs, targets)
+        # weight each loss by `self.weight_dict[<LOSS_NAME>]`
+        weighted_loss_dict: Dict[str, Tensor] = {k: v * self.weight_dict[k] for k, v in loss_dict.items()}
+        return weighted_loss_dict
+
+    def get_loss(self, loss_dict: Dict[str, Tensor]) -> Tensor:
+        # probably an awkward way to reduce it
+        return torch.tensor(list(loss_dict.values()), dtype=torch.float).sum()
+
+
+@dataclass
+class MaskFormerForSemanticSegmentationOutput(MaskFormerOutput):
+    segmentation: torch.FloatTensor = None
+
+
+class MaskFormerForSemanticSegmentation(nn.Module):
+    def __init__(self, config: MaskFormerConfig):
+        super().__init__()
+        self.model = MaskFormerModel(config)
+
+    def forward(
+        self,
+        pixel_values: Tensor,
+    ):
+        outputs: MaskFormerOutput = self.model(pixel_values)
+        # mask classes has shape [BATCH, QUERIES, CLASSES + 1]
+        # remove the null class `[..., :-1]`
+        mask_classes: Tensor = outputs.pred_logits.softmax(dim=-1)[..., :-1]
+        # mask probs has shape [BATCH, QUERIES, HEIGHT, WIDTH]
+        mask_probs: Tensor = outputs.pred_masks.sigmoid()
+        # now we want to sum over the queries,
+        # $ out_{c,h,w} =  \sum_q p_{q,c} * m_{q,h,w} $
+        # where $ softmax(p) \in R^{q, c} $ is the mask classes
+        # and $ sigmoid(m) \in R^{q, h, w}$ is the mask probabilities
+        # b(atch)q(uery)c(lasses), b(atch)q(uery)h(eight)w(idth)
+        segmentation: Tensor = torch.einsum("bqc, bqhw -> bchw", mask_classes, mask_probs)
+
+        return MaskFormerForSemanticSegmentationOutput(segmentation=segmentation, **outputs)
+
+
+class PanopticSegmentationSegment(TypedDict):
+    id: int
+    category_id: int
+    is_thing: bool
+    label: str
+
+
+@dataclass
+class MaskFormerForPanopticSegmentationOutput(MaskFormerForSemanticSegmentationOutput):
+    segments: List[PanopticSegmentationSegment] = None
```

Comment:
Also here, please don't overwrite output classes, and place at the top of the modeling file.